test/import/FSHImporter.context.test.ts

```diff
@@ -0,0 +1,398 @@
+    });
+
+    it('should parse child rules that have a blank path', () => {
+      const input = `
+      Profile: Foo
+      Parent: Patient
+      * name 1..1
+        * ^short = "foo"
+        * obeys inv1
+    `;
+
+      const result = importSingleText(input, 'Context.fsh');
+      expect(loggerSpy.getAllMessages('error')).toHaveLength(0);
+      expect(loggerSpy.getAllMessages('warn')).toHaveLength(0);
+      expect(result.profiles.size).toBe(1);
+      const profile = result.profiles.get('Foo');
+      expect(profile.name).toBe('Foo');
+      expect(profile.parent).toBe('Patient');
+      expect(profile.rules.length).toBe(3);
+      assertCardRule(profile.rules[0], 'name', 1, 1);
+      assertCaretValueRule(profile.rules[1], 'name', 'short', 'foo', false);
+      assertObeysRule(profile.rules[2], 'name', 'inv1');
+    });
+
+    it('should log an error when a rule is indented below a rule without a path', () => {
+      const input = `
+      Profile: Foo
+      Parent: Patient
+      * ^url = "http://example.org"
+        * name 1..1
+    `;
+
+      const result = importSingleText(input, 'Context.fsh');
+      expect(loggerSpy.getAllMessages('error')).toHaveLength(1);
+      expect(loggerSpy.getLastMessage('error')).toMatch(/Rule cannot be indented/);
+      expect(loggerSpy.getAllMessages('warn')).toHaveLength(0);
+      expect(result.profiles.size).toBe(1);
+      const profile = result.profiles.get('Foo');
+      expect(profile.name).toBe('Foo');
+      expect(profile.parent).toBe('Patient');
+      expect(profile.rules.length).toBe(2);
+      assertCaretValueRule(profile.rules[0], '', 'url', 'http://example.org', false);
+      assertCardRule(profile.rules[1], 'name', 1, 1);
+    });
+
+    it('should log an error when a rule is indented an invalid amount of spaces', () => {
+      // only 1 space of indent
+      const input = `
+        Instance: Foo
+        InstanceOf: Patient
+        * name.family = "foo"
+         * id = "bar"
+      `;
+
+      const result = importSingleText(input, 'Context.fsh');
+      expect(loggerSpy.getAllMessages('error')).toHaveLength(1);
+      expect(loggerSpy.getLastMessage('error')).toMatch(/Unable to determine context.*1 space/);
+      expect(loggerSpy.getAllMessages('warn')).toHaveLength(0);
+      expect(result.instances.size).toBe(1);
+      const instance = result.instances.get('Foo');
+      expect(instance.name).toBe('Foo');
+      expect(instance.instanceOf).toBe('Patient');
+      expect(instance.rules.length).toBe(2);
+      assertAssignmentRule(instance.rules[0], 'name.family', 'foo');
+      // rule is not assigned any context
+      assertAssignmentRule(instance.rules[1], 'id', 'bar');
+    });
+
+    it('should log an error when a rule is indented a negative amount of spaces', () => {
+      // only 1 space of indent
+      const input = `
+        Instance: Foo
+        InstanceOf: Patient
+          * name.family = "foo"
+        * id = "bar"
```

Comment:
No, everything is relative to the indent level of the first rule. That sets the `baseIndent`. If `id` was instead indented two spaces as well, this would be valid. That way someone can indent an entire definition if they want to.

Reply:
Ah.  I don't think we every said how that should work...  I'd actually like to discuss this with the team, because I'm not sure if it's the way we want it to work or not.  Personally, I find it a little confusing.

I might find it less confusing if the lines above were also indented to that level (e.g., the first line of the definition [e.g., `Instance:`] sets the original indent level) -- but it feels weird to me to have the definition start with 0 indent, then have the first rule establish a _different_ starting indent.

I kind of feel like if space before `*` is significant, that it should always be significant and always mean the same thing.  I also feel like supporting indent on the first rule might lead people to (wrongly) think something like this should work:
```
Profile: MyPatient
Parent: Patient
* name 1..* MS
* insert NameDetails

RuleSet: NameDetails
  * given 1..*
  * family 1..1
```

Anyway, I suggest we talk about this some more.

Reply:
Although I could go either way on the spaces before `*` on the first rule, I feel like that example is going to be a confusing case no matter what. Either way, people will need to just understand that the context gets applied before rules are inserted.

Reply:
We originally were proposing that a `RuleSet`'s first rule cannot ever be indented -- which is a little more clearly stated and enforced if we don't support optional indenting on the first rule in a definition.  But I agree... people may try to do it either way.  It's just a matter of what error they get:
> error: The first rule in a RuleSet cannot be indented.
or
> error: Cannot find element at path: given